client_options.go

```diff
@@ -33,66 +33,83 @@ type httpClientConfig struct {
 	timeout                     time.Duration
 }
 
+// WithProxyUrl configures a HTTP client to use the specified proxy URL.
+//
+// proxyUrl should be formatted as:
+//
+//	"http://user:pass@host:port"
 func WithProxyUrl(proxyUrl string) HttpClientOption {
 	return func(config *httpClientConfig) {
 		config.proxyUrl = proxyUrl
 	}
 }
 
+// WithCookieJar configures a HTTP client to use the specified cookie jar.
 func WithCookieJar(jar http.CookieJar) HttpClientOption {
 	return func(config *httpClientConfig) {
 		config.cookieJar = jar
 	}
 }
 
+// WithTimeout configures a HTTP client to use the specified request timeout.
+//
+// timeout is the request timeout in seconds.
 func WithTimeout(timeout int) HttpClientOption {
 	return func(config *httpClientConfig) {
 		config.timeout = time.Second * time.Duration(timeout)
 	}
 }
 
+// WithNotFollowRedirects configures a HTTP client to not follow HTTP redirects.
 func WithNotFollowRedirects() HttpClientOption {
 	return func(config *httpClientConfig) {
 		config.followRedirects = false
 	}
 }
 
+// WithRandomTLSExtensionOrder configures a TLS client to randomize the order of TLS extensions being sent in the ClientHello.
```

Comment:
GREASE and Padding Extension will stay in place. Only all other extensions will be randomized in order